pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -176,53 +177,60 @@ func (c *HPAWatcherClient) Start() {
 				if !leaderEngine.IsLeader() {
 					continue
 				}
+				// Updating the metrics against Datadog should not affect the HPA pipeline.
+				// If metrics are temporarily unavailable for too long, they will become `Valid=false` and won't be evaluated.
 				c.updateCustomMetrics()
 			}
 		}
 	}()
 }
 
-func (c *HPAWatcherClient) updateCustomMetrics() error {
+func (c *HPAWatcherClient) updateCustomMetrics() {
 	maxAge := int64(c.externalMaxAge.Seconds())
+
 	configMap, err := c.clientSet.CoreV1().ConfigMaps(c.ns).Get(datadogHPAConfigMap, metav1.GetOptions{})
 	if err != nil {
 		log.Infof("Error while retrieving the Config Map %s", datadogHPAConfigMap)
-		return nil
+		return
 	}
+
 	data := configMap.Data
 	if len(data) == 0 {
 		log.Debugf("No External Metrics to evaluate at the moment")
-		return nil
+		return
 	}
+
 	for _, d := range data {
 		cm := &CustomExternalMetric{}
-
 		json.Unmarshal([]byte(d), &cm)
 
-		if metav1.Now().Unix()-cm.Timestamp > maxAge {
-
+		if metav1.Now().Unix()-cm.Timestamp > maxAge || !cm.Valid {
+			cm.Valid = false
 			cm.Timestamp = metav1.Now().Unix()
 			cm.Value, err = QueryDatadogExternal(cm.Name, cm.Labels)
+
```

Comment:
Is it possible to call `validateMetric` here too? The code looks similar in both these places.